src/grdfill.c

```diff
@@ -599,8 +599,8 @@ EXTERN_MSC int GMT_grdfill (void *V_API, int mode, void *args) {
 	ID = gmt_M_memory_aligned (GMT, NULL, Grid->header->size, char);
 	/* Set the top and bottom boundary rows to UINT_MAX */
 	offset = (uint64_t)(Grid->header->pad[YHI] + Grid->header->n_rows) * Grid->header->mx;
-	for (node = 0; node < (uint64_t)Grid->header->pad[YHI]*Grid->header->mx; node++) ID[node] = ID[node+offset] = 1;
```

Comment:
@PaulWessel, is it necessary to set all of the boundary rows to 1, rather than just the outside most rows/columns?

Reply:
Not sure, will look to see why.

Reply:
Remember to change the **UINT_MAX** to 1 since I clearly changed my mind mid-coding and now use a char array.

Since we are only expanding (in grdfill_trace_the_hole) in the 4 cardinal directions it is hard to see why we would need more than one boundary of ones.  So you could change it only do the pad[?]-1 row and col inner pad and check.

Reply:
OK, I will update this for both the top/bottom and left/right boundaries.

Would using 1 be an issue in the odd case that someone uses grdfill -N1?

Reply:
No, the ID array is just used for keeping track of the NaNs, The -N affects what we look for, but you will see the code then simply replace grid values equal what -N says wiht NaNs since that is how the algo works,